alicloud/resource_alicloud_ons_instance.go

```diff
@@ -0,0 +1,192 @@
+	d.Set("instance_id", response.InstanceBaseInfo.InstanceId)
+	d.Set("instance_type", response.InstanceBaseInfo.InstanceType)
+	d.Set("instance_status", response.InstanceBaseInfo.InstanceStatus)
+	d.Set("release_time", time.Unix(int64(response.InstanceBaseInfo.ReleaseTime)/1000, 0).Format("2006-01-02 03:04:05"))
+
+	return nil
+}
+
+func resourceAlicloudONSInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	onsService := OnsService{client}
+
+	attributeUpdate := false
+
+	request := ons.CreateOnsInstanceUpdateRequest()
+	request.InstanceId = d.Get("instance_id").(string)
+	request.PreventCache = onsService.GetPreventCache()
+
+	if d.HasChange("name") {
+		d.SetPartial("name")
+		var name string
+		if v, ok := d.GetOk("name"); ok {
+			name = v.(string)
+		} else {
+			return fmt.Errorf("cann't change name to empty string")
+		}
+		request.InstanceName = name
+		attributeUpdate = true
+	}
+
+	if d.HasChange("remark") {
+		d.SetPartial("remark")
+		var remark string
+		if v, ok := d.GetOk("remark"); ok {
+			remark = v.(string)
+		} else {
+			return fmt.Errorf("cann't change remark to empty string")
+		}
+		request.Remark = remark
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		raw, err := onsService.client.WithOnsClient(func(onsClient *ons.Client) (interface{}, error) {
+			return onsClient.OnsInstanceUpdate(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw)
+	}
+
+	return resourceAlicloudONSInstanceRead(d, meta)
+}
+
+func resourceAlicloudONSInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	onsService := OnsService{client}
+
+	request := ons.CreateOnsInstanceDeleteRequest()
+	request.InstanceId = d.Get("instance_id").(string)
+	request.PreventCache = onsService.GetPreventCache()
+
+	raw, err := onsService.client.WithOnsClient(func(onsClient *ons.Client) (interface{}, error) {
+		return onsClient.OnsInstanceDelete(request)
+	})
+	if err != nil {
+		if IsExceptedErrors(err, []string{OnsInstanceNotExist}) {
+			return nil
+		}
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
+
+	return WrapError(onsService.WaitForOnsInstance(d.Id(), Deleted, DefaultTimeoutMedium))
+}
+
+func validateOnsInstanceName(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if len(value) < 3 || len(value) > 64 {
+		errors = append(errors, fmt.Errorf("%q cannot be less than 3 and longer than 64 characters", k))
+	}
+	return
+}
+
+func validateOnsInstanceRemark(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if len(value) > 128 {
+		errors = append(errors, fmt.Errorf("%q cannot be longer than 128 characters", k))
+	}
+	return
+}
```

Comment:
Please move the func `validateOnsInstanceName ` and `validateOnsInstanceRemark ` to validator.go